src/algebraic_topology/alternating_face_map_complex.lean

```diff
@@ -0,0 +1,233 @@
+  rw [← eq_neg_iff_add_eq_zero, ← finset.sum_neg_distrib],
+  /- we are reduced to showing that two sums are equal, and this is obtained
+  by constructing a bijection φ : S -> Sᶜ, which maps (i,j) to (j,i+1),
+  and by comparing the terms -/
+  let φ : Π (ij : P), ij ∈ S → P := λ ij hij,
+    (fin.cast_lt ij.2
+      (lt_of_le_of_lt (finset.mem_filter.mp hij).right (fin.is_lt ij.1)), ij.1.succ),
+  apply finset.sum_bij φ,
+  { -- φ(S) is contained in Sᶜ
+    intros ij hij,
+    simp only [finset.mem_univ, finset.compl_filter, finset.mem_filter, true_and,
+      fin.coe_succ, fin.coe_cast_lt] at hij ⊢,
+    linarith, },
+  { /- identification of corresponding terms in both sums -/
+    rintro ⟨i, j⟩ hij,
+    let jj : fin (n+2) := (φ (i,j) hij).1,
+    simp only [finset.mem_filter, finset.mem_univ, true_and] at hij,
+    simp only [term, d_l, d_r, φ],
+    simp only,
+    clear term d_l d_r,
+    repeat { rw [comp_zsmul, zsmul_comp], },
+    rw [← neg_smul, ← mul_smul, ← mul_smul],
+    rw [← show (-1 : ℤ)^(i : ℕ) * (-1 : ℤ)^(j : ℕ) =
+        - (-1 : ℤ)^(jj : ℕ) * (-1 : ℤ)^(i.succ : ℕ), by
+      { simp only [fin.coe_succ, fin.coe_cast_lt], ring_exp, }],
+    apply congr_arg,
+    have ineq : jj ≤ i, { rw ← fin.coe_fin_le, exact hij, },
+    rw category_theory.simplicial_object.δ_comp_δ X ineq,
+    simp only [fin.cast_succ_cast_lt], },
+  { -- φ : S → Sᶜ is injective
+    rintro ⟨i, j⟩ ⟨i', j'⟩ hij hij' h,
+    rw [prod.mk.inj_iff],
+    split,
+    { have h1 := congr_arg prod.snd h,
+      simp only [fin.succ_inj] at h1,
+      exact congr_arg _ h1 },
+    { have h1 := congr_arg fin.cast_succ (congr_arg prod.fst h),
+      simp only [fin.cast_succ_cast_lt] at h1,
+      exact h1, }, },
+  { -- φ : S → Sᶜ is surjective
+    clear term d_l d_r,
+    rintro ⟨i', j'⟩ hij',
+    simp only [true_and, finset.mem_univ, finset.compl_filter, not_le,
+      finset.mem_filter] at hij',
+    refine ⟨(j'.pred _,fin.cast_succ i'),_,_⟩,
+    { intro H,
+      rw [H] at hij',
+      simp only [nat.not_lt_zero, fin.coe_zero] at hij',
+      exact hij', },
+    { simp only [true_and, finset.mem_univ, fin.coe_cast_succ, fin.coe_pred,
+        finset.mem_filter],
+      exact nat.le_pred_of_lt hij', },
+    { simp only [prod.mk.inj_iff, fin.succ_pred, fin.cast_lt_cast_succ],
+      split; refl, }, },
+end
+
+/-!
+## Construction of the alternating face map complex functor
+-/
+
+/-- The alternating face map complex, on objects -/
+def obj : chain_complex C ℕ := chain_complex.of (λ n, X _[n]) (obj_d X) (d_squared X)
+
+variables {X} {Y}
+
+/-- The alternating face map complex, on morphisms -/
+@[simp]
+def map (f : X ⟶ Y) : obj X ⟶ obj Y :=
+chain_complex.of_hom _ _ _ _ _ _
+  (λ n, f.app (op [n]))
+  (λ n,
+    begin
+      repeat { rw obj_d },
+      rw [comp_sum, sum_comp],
+      apply congr_arg,
+      ext,
+      rw [comp_zsmul, zsmul_comp],
+      apply congr_arg,
+      erw f.naturality,
+      refl,
```

Comment:
This can be golfed to

```lean
      dsimp,
      rw [comp_sum, sum_comp],
      apply finset.sum_congr rfl (λ x h, _),
      rw [comp_zsmul, zsmul_comp],
      apply congr_arg,
      erw f.naturality,
      refl,
```